public/reorderSiteAwards.php

```diff
@@ -10,46 +10,171 @@
+
+            if (draggedRowIndex !== dropTargetIndex) {
+                if (draggedRowIndex < dropTargetIndex) {
+                    dropTarget.parentNode.insertBefore(currentGrabbedRow, dropTarget.nextSibling);
+                } else {
+                    dropTarget.parentNode.insertBefore(currentGrabbedRow, dropTarget);
+                }
+
+                updateDisplayOrderValues(dropTableId, initialDisplayOrderValue);
+            }
+        }
+    }
+
+    dropTarget.classList.remove('border');
+    dropTarget.classList.remove('border-menu-link');
+}
+
+function getInitialDisplayOrderValue(dropTableId) {
+    let initialDisplayOrderValue = 0;
+
+    const tableRows = document.querySelectorAll(`#${dropTableId} tbody tr`);
+    if (tableRows[0]) {
+        const firstRowOrderInput = tableRows[0].querySelector('.displayorderedit');
+        initialDisplayOrderValue = parseInt(firstRowOrderInput.value, 10);
+    }
+
+    return initialDisplayOrderValue;
+}
+
+function updateDisplayOrderValues(dropTableId, startingValue = 0) {
+    const tableRows = document.querySelectorAll(`#${dropTableId} tbody tr`);
+
+    tableRows.forEach((row,index) => {
+        const displayOrderInput = row.querySelector('.displayorderedit');
+        const currentValue = parseInt(displayOrderInput.value, 10);
+
+        if (currentValue !== -1) {
+            displayOrderInput.value = startingValue + (10 * index);
+        }
     });
 }
 </script>
 <div id="mainpage">
     <div id="leftcontainer">
         <?php
         echo "<h2>Reorder Site Awards</h2>";
-        echo "<a class='btn btn-link' href='/reorderSiteAwards.php'>Refresh Page</a><br>";
 
-        echo "<p class='embedded'><b>Instructions:</b> These are your site awards as displayed on your user page. " .
-            "The awards will be ordered by 'Display Order', the column found on the right, in order from smallest to greatest. " .
-            "Adjust the numbers on the right to set an order for them to appear in. Setting a 'Display Order' value to -1 " .
-            "will hide the site award. Any changes you make on this page will instantly " .
-            "take effect on the website, but you will need to press 'Refresh Page' to see the new order on this page. " .
-            "The right panel represents how the site awards will look on your user page.</p>";
+        echo <<<HTML
+            <div class="embedded grid gap-y-4">
+                <p>
+                    To rearrange your site awards, modify the 'Display Order' value in the rightmost 
+                    column of each award row. The awards will be displayed on your user page in 
+                    ascending order based on these values. To hide an award, change its 'Display Order' 
+                    value to -1. Remember to save your changes by clicking the 'Save' button, and 
+                    your updates will appear on your user page immediately.
+                </p>
```

Comment:
You may want to mention that the rows can be dragged to reorder them, as that's the feature being introduced.

Reply:
This text very much no longer aligns with the implementation.

Reply:
This text has been rewritten.

I have heard a saying that good UX is like a good joke, it doesn't have to be explained to the person for them to understand it. Ideally I would like to get to the point that this text can be removed altogether, but I'm just not sure if we're there yet.